py_src/ipyelk/nx/transformer.py

```diff
@@ -235,23 +324,34 @@ def get_node_size(self, node: ElkNode) -> Tuple[Optional[float], Optional[float]
             width = self.text_scale
         return width, height
 
-    def make_labels(self, node) -> Optional[List[ElkLabel]]:
+    async def size_label(self, text: str) -> TextSize:
+        if self.text_sizer is None:
+            return TextSize(
+                width=self.text_scale * len(text),
+                height=10,  # TODO add height default
+            )
+        else:
+            return await self.text_sizer.measure(text)
+
+    async def make_labels(self, node) -> Optional[List[ElkLabel]]:
         if node is None:
             return None
         g, tree = self.source
         data = g.nodes[node]
         name = data.get(self.label_key, data.get("_id", f"{node}"))
-        width = self.text_scale * len(name)
-
-        return [
-            ElkLabel(
-                id=f"{name}_label_{node}",
-                text=name,
-                width=width,
-                x=self.label_offset,
-                y=self.label_offset,
-            )
-        ]
+        size = await self.size_label(name)
+        # width = self.sizer.measure(name)
+        label = ElkLabel(
+            id=f"{name}_label_{node}",
+            text=name,
+            width=size.width,
+            height=size.height,
+            # x=self.label_offset,
+            # y=self.label_offset,
```

Comment:
these staying in?

Reply:
I'll take a look at these on https://github.com/jupyrdf/ipyelk/issues/19.